var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -113,11 +113,14 @@ def mpi_dependent_options(self):
                 raise RuntimeError('\n'.join(errors))
         else:
             compiler_opts = [
-                '--with-cpp=cpp',
-                '--with-cxxcpp=cpp',
                 '--with-mpi=1',
                 '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
             ]
+        if sys.platform != "darwin":
```

Comment:
The problem is that Spack's cpp doesn't work. I'm building first gcc (6.2) via Spack, then use this gcc to build other packages. Apparently gcc's cpp doesn't work on Sierra with certain system header files. Similar issues have existed with gcc on Darwin for a long time, so I didn't investigate details but instead simply reverted to letting PETSc choose the cpp it naturally wants. I don't have the exact error message handy any more.